src/lib.rs

```diff
@@ -205,12 +212,39 @@ pub const NAMESPACE_X500: Uuid = Uuid {
             0x80, 0xb4, 0x00, 0xc0, 0x4f, 0xd4, 0x30, 0xc8]
 };
 
+/// The number of 100 ns ticks between 
+/// the UUID epoch 1582-10-15 00:00:00 and the Unix epoch 1970-01-01 00:00:00
+#[cfg(feature = "v1")]
+const UUID_TICKS_BETWEEN_EPOCHS : u64 = 0x01B21DD213814000;
 
 /// An adaptor for formatting a `Uuid` as a URN string.
 pub struct Urn<'a> {
     inner: &'a Uuid,
 }
 
+/// A stateful context for the v1 generator to help ensure process-wide uniqueness
+#[cfg(feature = "v1")]
+pub struct UuidV1Context {
+    count: AtomicUsize,
+}
+
+#[cfg(feature = "v1")]
+impl UuidV1Context {
+
+    /// Creates a thread-safe, internally mutable context to help ensure uniqueness
+    ///
+    /// This is a context which can be shared across threads.  It maintains an internal
+    /// counter that is incremented at every request, the value ends up in the clock_seq
+    /// portion of the V1 uuid (the fourth group).  This will improve the probability
+    /// that the UUID is unique across the process.
+    pub fn new(seed : Option<u64>) -> UuidV1Context {
+        let initval = seed.unwrap_or(thread_rng().gen());
```

Comment:
True yeah. I wonder, how about something like this?

```rust
impl UuidV1Context {
    pub fn new(seed: u16) -> UuidV1Context;
}

impl Rand for UuidV1Context {
    // ...
}
```

That way they can be both explicitly requested? And we can document in `new` that you may wish to use `rand::random::<UuidV1Context>()` to generate a new context?